### PR TITLE
Break the routes

### PR DESCRIPTION
**Pull request by devvoh**
Arrays are stupid. Now that there's `->get()`, `->post()`, etc. available on `\Parable\Framework\App`, it's much better to use those, especially since, as methods, they offer type hinting whereas the array method is far too loose and fragile.

### CODE REVIEW

**Comment by devvoh** [comment]
Note: this completely borks named routes (and therefore all methods that work with routes based on their name, which happen to be fantastic methods), so a further commit which fixes that is coming later today.

This will add `$name = null` to the parameter list for the simple routing methods before `$templatePath`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ __Backwards-incompatible Changes__
...
 - `\Parable\Routing\Route` no longer supports `template` for the template path, but the more correctly named `templatePath` instead. Because of this, it now checks more strictly whether valid properties are set through the Routing array. `setDataFromArray()` attempts to call setters named like the properties. Any that are not available with a setter will throw an Exception. All properties are now also `protected`.
 - `\Parable\Routing\Router` now also supports adding a completely set-up `Route` object directly (or in an array), without having to pass them as arrays, through `addRoute()` and `addRoutes()`. These methods already existed, but those are now renamed to `addRouteFromArray()` and `addRoutesFromArray()`.
+- The default structure's `\Config\App` is now expected to extend the abstract base class `\Parable\Framework\Routing\AbstractRouting` and the old `Interfaces\Routing` interface has been removed. **This removes passing routes as arrays!**. See the new implementation, using the method calls on `\Parable\Framework\App` in the structure's `app/Routing/App.php_struct`.
```
Comment (dmvdbrugge) [inline CHANGELOG.md]: Shouldn't this line start with the following?
> The default structure's `\Routing\App`

(As seen in the struct file)

File: structure/app/Routing/App.php_struct [hunk 2]
```diff
@@ -2,34 +2,36 @@
...
+        /*
+         * By calling ->any(), you're saying that the method doesn't matter. As long as Parable accepts the method,
+         * it'll match if the url does.
+         */
+        $this->app->any("/test/{id}/{name}", [\Controller\Home::class, "test"]);
+
+        /*
+         * It's also possible to specify multiple methods to respond to. Also, callables.
+         */
+        $this->app->multiple(["GET", "POST"], "/callable", function () {
```
Comment (dmvdbrugge) [inline structure/app/Routing/App.php_struct]: Use `Request::METHOD_*`?

File: structure/app/Routing/App.php_struct [hunk 3]
```diff
@@ -2,34 +2,36 @@
...
-             */
-            'test-parameters' => [
-                'methods' => ['GET'],
-                'url' => '/test/{id}/{name}',
-                'controller' => \Controller\Home::class,
-                'action' => 'test',
-            ],
-        ];
+        /*
+         * Valid methods are GET/POST/PUT/PATCH/DELETE/OPTIONS. If a url matches but the method doesn't,
```
Comment (dmvdbrugge) [inline structure/app/Routing/App.php_struct]: Reference `Request::METHOD_*` or `Request::VALID_METHODS` constants?

**Comment by devvoh** [comment]
@dmvdbrugge The tests still worked because it wasn't actively under test, but passively. That's why I've added some tests to check for both named routes and unnamed (which will generate a uniqid for it instead).

**Pull request merged**